optimum/exporters/onnx/__main__.py

```diff
@@ -221,6 +221,12 @@ def main_export(
             " and passing it is not required anymore."
         )
 
+    if task in ["stable-diffusion", "stable-diffusion-xl"]:
+        logger.warning(
+            f"The task `{task}` is deprecated and will be removed in a future release of Optimum. "
+            "Please use one of the following tasks instead: `text-to-image`, `image-to-image`, `inpainting`."
+        )
+
```

Comment:
We warn and then it gets replaced with `text-to-image` by default. This is the same as the previous behavior for `stable-diffusion` but not `stable-diffusion-xl` as the model loader in that case used to be an `image-to-image` pipeline.

Reply:
`_optional_components` was introduced in diffusers v0.22.0 https://github.com/huggingface/diffusers/blob/v0.22.0/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L144 for `StableDiffusionXLPipeline` (not the case for `StableDiffusionXLImg2ImgPipeline`). I think we might need to upgrade [DIFFUSERS_MINIMUM_VERSION](https://github.com/huggingface/optimum/blob/dedb852eaea3d3d899920f4017b9fd899eb2756b/optimum/utils/import_utils.py#L53) as well to avoid any potential issue when loading a SDXL model with missing optional components